### ISSUE TITLE
[RFC] Improve the way annotations are enabled

### ISSUE THREAD

**Issue by javiereguiluz** [symfony/recipes#118]
In a Symfony Flex app, I expected to enable annotations by running this:

```
$ composer require annotations
```

However, that installs the `symfony/annotation-pack`, which only installs the low-level libraries needed for annotations (doctrine/cache and doctrine/annotations) but it doesn't really enable annotations in the application.

---

First, this was a problem because it crashed my browser, [as reported here](https://github.com/symfony/symfony/issues/23566).

Second, even if this comment from `config/routing.yaml` is perfectly clear:

```
# Depends on sensio/framework-extra-bundle, doctrine/annotations, and doctrine/cache
#   install with composer req sensio/framework-extra-bundle annot
```

It's a bit disappointing that `composer require annotations` doesn't enable annotations and you must also execute `composer require sensio/framework-extra-bundle`

-----

Solutions:

* Maybe we should create a better recipe for annotations?
* Maybe we should add annotations support in frameworkbundle and forget about the framework extra bundle?

**Comment by Pierstoval** [comment]
`symfony/annotations-pack` is the base for annotations, but what you need is annotations for routing, so maybe a `routing-annotations` pack can be created, which will require `sensio/framework-extra-bundle` & `symfony/annotations-pack`?

**Comment by ogizanagi** [comment]
Note that thanks to https://github.com/symfony/symfony/pull/23044 and FQCN services, we already won't need the `sensio/framework-extra-bundle` anymore in 3.4, when flex will become the favored way to create a new symfony project.
I'm not sure we need anything else until this. As you said, the comment is self explanatory, and `symfony/annotations-pack` has nothing to do with routing. I don't think creating a new pack for this is worth it. WDYT?

**Comment by weaverryan** [comment]
@ogizanagi If I understand you correctly, on 3.4, the user would only need to run `composer require annotations` (and uncomment out the annotations route import) in order for `@Route` annotations to work? Is that correct?

Btw, if that's true, what about `@Method`? I only see this annotation in FWExtraBundle, not Symfony core.

Thanks :)

**Comment by ogizanagi** [comment]
> the user would only need to run composer require annotations (and uncomment out the annotations route import) in order for `@Route` annotations to work

True :)

---

> Btw, if that's true, what about `@Method`? I only see this annotation in FWExtraBundle, not Symfony core.

True again. the `@Method` annotation isn't part of the core. But it's juste a shortcut annotation; you can already configure the methods within the `@Route` annotation (`@Route("foo", methods={"GET"})`), so it's probably not a reason good enough to install the FWExtraBundle.

**Comment by fabpot** [comment]
fixed by #179
